http/delete_handler.go

```diff
@@ -0,0 +1,207 @@
+		Logger:           b.Logger,
+
+		BucketService:       b.BucketService,
+		DeleteService:       b.DeleteService,
+		OrganizationService: b.OrganizationService,
+	}
+
+	h.HandlerFunc("POST", deletePath, h.handleDelete)
+	return h
+}
+
+func (h *DeleteHandler) handleDelete(w http.ResponseWriter, r *http.Request) {
+	span, r := tracing.ExtractFromHTTPRequest(r, "DeleteHandler")
+	defer span.Finish()
+
+	ctx := r.Context()
+	defer r.Body.Close()
+
+	a, err := pcontext.GetAuthorizer(ctx)
+	if err != nil {
+		h.HandleHTTPError(ctx, err, w)
+		return
+	}
+
+	dr, err := decodeDeleteRequest(
+		ctx, r,
+		h.OrganizationService,
+		h.BucketService,
+	)
+	if err != nil {
+		h.HandleHTTPError(ctx, err, w)
+		return
+	}
+
+	p, err := influxdb.NewPermissionAtID(dr.Bucket.ID, influxdb.WriteAction, influxdb.BucketsResourceType, dr.Org.ID)
+	if err != nil {
+		h.HandleHTTPError(ctx, &influxdb.Error{
+			Code: influxdb.EInternal,
+			Op:   "http/handleDelete",
+			Msg:  fmt.Sprintf("unable to create permission for bucket: %v", err),
+			Err:  err,
+		}, w)
+		return
+	}
+
+	if !a.Allowed(*p) {
+		h.HandleHTTPError(ctx, &influxdb.Error{
+			Code: influxdb.EForbidden,
+			Op:   "http/handleDelete",
+			Msg:  "insufficient permissions to delete",
+		}, w)
+		return
+	}
+
+	// send delete points request to storage
+	err = h.DeleteService.DeleteBucketRangePredicate(ctx,
+		dr.Org.ID,
+		dr.Bucket.ID,
+		dr.Start,
+		dr.Stop,
+		dr.Predicate,
+	)
+	if err != nil {
+		h.HandleHTTPError(ctx, err, w)
+		return
+	}
+	h.Logger.Debug("deleted",
+		zap.String("orgID", fmt.Sprint(dr.Org.ID.String())),
+		zap.String("buketID", fmt.Sprint(dr.Bucket.ID.String())),
+	)
+
+	w.WriteHeader(http.StatusNoContent)
+}
+
+func decodeDeleteRequest(ctx context.Context, r *http.Request, orgSvc influxdb.OrganizationService, bucketSvc influxdb.BucketService) (*deleteRequest, error) {
+	dr := new(deleteRequest)
+	err := json.NewDecoder(r.Body).Decode(dr)
+	if err != nil {
+		return nil, &influxdb.Error{
+			Code: influxdb.EInvalid,
+			Msg:  "invalid request; error parsing request json",
+			Err:  err,
+		}
+	}
+	if dr.Org, err = queryOrganization(ctx, r, orgSvc); err != nil {
+		return nil, err
+	}
+
+	if dr.Bucket, err = queryBucket(ctx, r, bucketSvc); err != nil {
+		return nil, err
+	}
+	return dr, nil
+}
+
+type deleteRequest struct {
+	Org       *influxdb.Organization
+	Bucket    *influxdb.Bucket
+	Start     int64
+	Stop      int64
+	Predicate influxdb.Predicate
+}
+
+type deleteRequestDecode struct {
+	Start     string          `json:"start"`
+	Stop      string          `json:"stop"`
+	Predicate json.RawMessage `json:"predicate"`
+}
+
+func (dr *deleteRequest) UnmarshalJSON(b []byte) error {
+	var drd deleteRequestDecode
+	err := json.Unmarshal(b, &drd)
+	if err != nil {
+		return &influxdb.Error{
+			Code: influxdb.EInvalid,
+			Msg:  "Invalid delete predicate node request",
+			Err:  err,
+		}
+	}
+	*dr = deleteRequest{}
+	start, err := time.Parse(time.RFC3339Nano, drd.Start)
```

Comment:
this might be nice to pull out into a resusable type, something like https://play.golang.org/p/-UH4K1WIJKW . Just a suggestion, there is probably better ways 🤷‍♂, definitely not mandatory

Reply:
yeah, i like the reusable time type idea. I think we have several places like that already. Maybe we should create a new issue to track and convert them all.